src/dfx/src/commands/deps/deploy.rs

```diff
@@ -0,0 +1,105 @@
+use crate::lib::deps::InitJson;
+use crate::lib::deps::{
+    get_pull_canisters_in_config, get_pulled_wasm_path, load_init_json, load_pulled_json,
+    validate_pulled,
+};
+use crate::lib::environment::Environment;
+use crate::lib::error::DfxResult;
+use crate::lib::root_key::fetch_root_key_if_needed;
+
+use anyhow::{anyhow, Context};
+use candid::Principal;
+use clap::Parser;
+use fn_error_context::context;
+use ic_agent::Agent;
+use ic_utils::interfaces::{management_canister::builders::InstallMode, ManagementCanister};
+use slog::{info, Logger};
+
+/// Deploy pulled canisters.
+#[derive(Parser)]
+pub struct DepsDeployOpts {
+    /// Specify the canister to deploy. You can specify its name (as defined in dfx.json) or Principal.
+    /// If not specified, all pulled canisters will be deployed.
+    canister: Option<String>,
+}
+
+pub async fn exec(env: &dyn Environment, opts: DepsDeployOpts) -> DfxResult {
+    let logger = env.get_logger();
+    let project_root = env.get_config_or_anyhow()?.get_project_root().to_path_buf();
+    let pulled_json = load_pulled_json(&project_root)?;
+    let pull_canisters_in_config = get_pull_canisters_in_config(env)?;
+    validate_pulled(&pulled_json, &pull_canisters_in_config)?;
+
+    let init_json = load_init_json(&project_root)?;
+
+    fetch_root_key_if_needed(env).await?;
+    let agent = env
+        .get_agent()
+        .ok_or_else(|| anyhow!("Cannot get HTTP client from environment."))?;
+
+    match opts.canister {
+        Some(canister) => {
+            let canister_id = match pull_canisters_in_config.get(&canister) {
+                Some(canister_id) => *canister_id,
+                None => Principal::from_text(canister).with_context(|| {
+                    "The canister is not a valid Principal nor a name specified in dfx.json"
+                })?,
+            };
+            create_and_install(agent, logger, &canister_id, &init_json).await?;
+        }
+        None => {
+            for canister_id in pulled_json.canisters.keys() {
+                create_and_install(agent, logger, canister_id, &init_json).await?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+#[context("Failed to create and install canster {}", canister_id)]
+async fn create_and_install(
+    agent: &Agent,
+    logger: &Logger,
+    canister_id: &Principal,
```

Comment:
Should this somewhere also install dependencies of the pulled canister?  Or is that for later work?

Reply:
`dfx deps pull` and `dfx deps init` fetches all direct and indirect dependencies and set their init arguments.
Then `dfx deps deploy` can deploy them in arbitrary order. There is no requirement that the dependency should be deployed before the dependent.

It's possible that the user forget to deploy some indirect dependencies. When they make calls that reach the dependencies, the replica will error that the canister is not installed.